src/UICrafter.Client/Components/Properties/Components/TextboxProperties.razor

```diff
@@ -0,0 +1,42 @@
+<MudTextField @bind-Value="SelectedComponent.Textbox.Label" Label="Textbox label" Variant="Variant.Outlined" />
+<MudTextField @bind-Value="SelectedComponent.Textbox.NumberOfLines" Label="Number of lines" Variant="Variant.Outlined" />
+<MudSelect T="string" Label="Source Reference" @bind-Value="SelectedComponent.Textbox.SourceRef">
+    <MudSelectItem Value="Guid.Empty.ToString()">None</MudSelectItem>
+    @foreach (var component in UIComponents.Where(c => c.ComponentCase == UIComponent.ComponentOneofCase.Button))
+    {
+        <MudSelectItem Value="@component.Guid">@component.Button.Label</MudSelectItem>
+    }
+</MudSelect>
+
+<MudField Label="Json field or query" Variant="Variant.Outlined" InnerPadding="false">
+    <MudRadioGroup T="bool" Value=@SelectedComponent.Textbox.IsJsonQuery ValueChanged="RadioChanged">
+        <MudRadio Value="false">Field</MudRadio>
+        <MudRadio Value="true">Query</MudRadio>
+    </MudRadioGroup>
+</MudField>
+
+@if (SelectedComponent.Textbox.IsJsonQuery)
+{
+    <MudTextField @bind-Value="SelectedComponent.Textbox.JsonField" Label="Json Query" Variant="Variant.Outlined" />
+}
+else
+{
+    <MudTextField @bind-Value="SelectedComponent.Textbox.JsonField" Label="Json Field" Variant="Variant.Outlined" />
+}
+
+
+@code {
+    [CascadingParameter]
+    public required UIComponent SelectedComponent { get; set; }
+
+    [Parameter]
+    public required ObservableCollection<UIComponent> UIComponents { get; set; }
+
+    private void RadioChanged(bool isQuery)
+    {
+        SelectedComponent.Textbox.IsJsonQuery = isQuery;
+
+        (SelectedComponent.Textbox.JsonField) = (SelectedComponent.Textbox.JsonField);
+        Log.Debug(SelectedComponent.Textbox.JsonField + ", " + SelectedComponent.Textbox.IsJsonQuery);
```

Comment:
BRUH what is this. Lav det om til en bind

Reply:
What do you want to change? Doing this RadioChanged is cleaner as the values needs to be switched?